### PR TITLE
Adding SNS To lambda DDK stage and SNS factory

### PR DESCRIPTION
**Pull request by anmolsgandhi**
### Feature
Adding SNS lambda stage and SNS factory

### Detail
- SNSToLambda DDK Stage 
- SNS Factory for best practices 
- tests for the above

### Relates
- #239 #240 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/aws_ddk_core/stages/sns_lambda.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                dead_letter_queue=SQSFactory.queue(
+                    self,
+                    id=f"{id}-dlq",
+                    environment_id=environment_id,
+                )
+                if dead_letter_queue_enabled
+                else None,
+                filter_policy=filter_policy,
+            )
+        )
```
Comment (LeonLuttenberger) [inline core/aws_ddk_core/stages/sns_lambda.py:170]: Should we have an SQS queue in between the SNS topic and the Lambda to make the lambda more efficient? Also, have you tested how the redrive  logic works with just an SNS queue? Is there an easy way to republish a message from the DLQ into the SNS topic?

**Review by malachi-constant — COMMENTED** [review]
I'm going to start converting your code to TS ( #250 ) tomorrow and we can move the convo over there once its drafted.

**Comment by malachi-constant** [comment]
Can we move discussion of this stage over to #252

**Comment by malachi-constant** [comment]
Closed and continued in #252

**Pull request closed without merging**